pkg/apiserver/watch.go

```diff
@@ -181,11 +182,20 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	flusher.Flush()
 
+	delay := wait.NewChannelDelayedAction(
+		wait.Interval{Count: 0, Interval: 5 * time.Millisecond},
+		wait.Interval{Count: 10, Interval: 100 * time.Millisecond},
+		wait.Interval{Count: 100, Interval: 500 * time.Millisecond},
+	)
+
 	buf := &bytes.Buffer{}
 	for {
 		select {
 		case <-cn.CloseNotify():
 			return
+		case <-delay.After():
+			delay.Done()
+			flusher.Flush()
```

Comment:
We should probably also flush at the end (e.g. if it finishes because of timeout).